cartographer_grpc/local_trajectory_uploader.cc

```diff
@@ -31,22 +32,25 @@ const cartographer::common::Duration kPopTimeout =
 
 LocalTrajectoryUploader::LocalTrajectoryUploader(
     const std::string &uplink_server_address)
-    : client_channel_(grpc::CreateChannel(uplink_server_address,
-                                          grpc::InsecureChannelCredentials())),
-      service_stub_(proto::MapBuilderService::NewStub(client_channel_)) {}
+    : client_channel_(grpc::CreateChannel(
+          uplink_server_address, grpc::InsecureChannelCredentials())) {}
 
 LocalTrajectoryUploader::~LocalTrajectoryUploader() {
-  if (imu_writer_.client_writer) {
-    CHECK(imu_writer_.client_writer->WritesDone());
-    CHECK(imu_writer_.client_writer->Finish().ok());
+  if (add_imu_client_) {
+    CHECK(add_imu_client_->WritesDone());
+    CHECK(add_imu_client_->Finish().ok());
+  }
+  if (add_odometry_client_) {
+    CHECK(add_odometry_client_->WritesDone());
+    CHECK(add_odometry_client_->Finish().ok());
   }
-  if (odometry_writer_.client_writer) {
-    CHECK(odometry_writer_.client_writer->WritesDone());
-    CHECK(odometry_writer_.client_writer->Finish().ok());
+  if (add_fixed_frame_pose_client_) {
```

Comment:
add_landmark_client_? (or add a todo and add it separately)

Reply:
done.